src/prpy/base/robot.py

```diff
@@ -288,107 +286,94 @@ def PostProcessPath(self, path, defer=False, executor=None,
-                    # before execution.
-                    else:
-                        if self.simplifier is not None:
-                            logger.debug('Shortcutting an unconstrained path.')
-                            shortcut_path = self.simplifier.ShortcutPath(
-                                cloned_robot, path, defer=False, **shortcut_options)
-                        else:
-                            logger.debug('Skipping shortcutting; no simplifier'
-                                         ' available.')
-                            shortcut_path = path
+            # Planners only operate on the active DOFs. We'll set any DOFs
+            # in the trajectory as active.
+            env = path.GetEnv()
+            cspec = path.GetConfigurationSpecification()
 
-                        logger.debug('Smoothing an unconstrained path.')
-                        traj = self.smoother.RetimeTrajectory(
-                            cloned_robot, shortcut_path, defer=False,
-                            **smoothing_options)
+            used_bodies = cspec.ExtractUsedBodies(env)
+            if self not in used_bodies:
+                raise ValueError(
+                    'Robot "{:s}" is not in the trajectory.'.format(
+                        self.GetName()))
 
-                # Copy the trajectory into the output environment.
-                output_traj = CopyTrajectory(traj, env=self.GetEnv()) 
+            # Extract active DOFs from teh trajectory and set them as active.
+            dof_indices, _ = cspec.ExtractUsedIndices(self)
 
-                # Copy meta-data from the path to the output trajectory.
-                output_traj.SetDescription(path.GetDescription())
+            if util.HasAffineDOFs(cspec):
+                affine_dofs = (DOFAffine.X | DOFAffine.Y | DOFAffine.RotationAxis)
 
-                return output_traj
+                # Bug in OpenRAVE ExtractUsedIndices function makes 
+                # dof_indices = affine_dofs. Temporary workaround for that bug.
+                dof_indices = []
+                logger.warning(
+                    'Trajectory contains affine DOFs. Any regular DOFs'
+                    ' will be ignored.'
+                )
+            else:
+                affine_dofs = 0
 
-        if defer is True:
-            from trollius.executor import get_default_executor
-            from trollius.futures import wrap_future
+            cloned_robot.SetActiveDOFs(dof_indices, affine_dofs)
+            logger.debug(
+                'Setting robot "%s" DOFs %s (affine? %d) as active for'
+                ' post-processing.',
+                cloned_robot.GetName(), list(dof_indices), affine_dofs
+            )
+
+            if len(dof_indices) and affine_dofs:
+                raise ValueError(
+                    'Trajectory contains both affine and regular DOFs.')
+            # Special case for timing affine-only trajectories.
```

Comment:
It looks a bit odd to have no newline between the `ValueError` and this comment. I suppose it's fine, though, because we do this all the time on `elif`s.